cmd/api-headers.go

```diff
@@ -64,6 +65,18 @@ func setCommonHeaders(w http.ResponseWriter) {
 
 // Encodes the response headers into XML format.
 func encodeResponse(response interface{}) []byte {
+	var bytesBuffer bytes.Buffer
+	bytesBuffer.WriteString(xml.Header)
+	buf, err := xml.Marshal(response)
+	if err != nil {
+		logger.LogIf(GlobalContext, err)
+		return nil
+	}
+	bytesBuffer.Write(buf)
+	return bytesBuffer.Bytes()
```

Comment:
Let's reduce the allocs and copies a bit:
```suggestion
	var buf bytes.Buffer
	buf.WriteString(xml.Header)
	if err := xml.NewEncoder(&buf).Encode(response); err != nil {
		logger.LogIf(GlobalContext, err)
		return nil
	}
	return buf.Bytes()
```